lib/Commands/BuildSystemCommand.cpp

```diff
@@ -405,13 +409,78 @@ static int executeParseCommand(std::vector<std::string> args) {
 
 class BasicBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
   std::unique_ptr<basic::FileSystem> fileSystem;
+
+  /// The previous SIGINT handler.
+  struct sigaction previousSigintHandler;
```

Comment:
Yep, it was saved but not restored. Now done in the destructor, along with closing the pipe.